src/librustc_typeck/check/mod.rs

```diff
@@ -1275,10 +1275,28 @@ fn compare_impl_method<'tcx>(tcx: &ty::ctxt<'tcx>,
             debug!("checking trait method for compatibility: impl ty {}, trait ty {}",
                    impl_fty.repr(tcx),
                    trait_fty.repr(tcx));
-            span_err!(tcx.sess, impl_m_span, E0053,
-                      "method `{}` has an incompatible type for trait: {}",
-                      token::get_name(trait_m.name),
-                      ty::type_err_to_str(tcx, &terr));
+
+            // Errors may be multi-line. An aligned multi-line error will have
+            // `expected` and `found` on consecutive lines. See issue 19464. A
+            // newline must be prepended so `expected` is located properly.
+            let ty_err = ty::type_err_to_str(tcx, &terr);
+            if ty_err.contains("\n") {
+                span_err!(tcx.sess, impl_m_span, E0053,
+                          "method `{}` has an incompatible type for trait:\n {}",
+                //                                                          ^
+                // Errors will have one of these 2 forms:
+                // ` expected error, ...
+                // `(expected error, ...)
+                // This is the first. It is padded so the error is aligned properly.
+                          token::get_name(trait_m.name),
+                          ty_err);
+            } else {
+                span_err!(tcx.sess, impl_m_span, E0053,
```

Comment:
This throws a warning I think because `E0053` was duplicated into a different branch. Not sure how to avoid it.